### PR TITLE
1.10: Add spelling check to CI and fix errors.

### PR DESCRIPTION
**Pull request by byrnHDF**
Compared spelling changes to 1.14.

### CODE REVIEW

**Comment by brtnfld** [comment]
You might try ./.github/<...> in the yml file.

**Comment by byrnHDF** [comment]
> You might try ./.github/<...> in the yml file.

That fixed the codespell file being included but it seems that the "Ot" addition to ignored_words_list is being ignored!

File: bin/batch/ctest.qsub.in.cmake [hunk 1]
```diff
@@ -11,11 +11,13 @@ CTEST_CMD=`which ctest`
 
 cd @HDF5_BINARY_DIR@
 if [[ $SUMMARY_FILE == *"ctestS"* ]]; then
-  CMD="${CTEST_CMD} . -E MPI_TEST_ -C Release -j 32 -T test"
+  CMD="${CTEST_CMD} -S ctest_serial.cmake"
+  qsub -t 60 -n 1 -q debug-flat-quad -A ${ACCOUNT_ID} ${CMD} >& ${SUMMARY_FILE}
+  echo "Done running ctest serial command."
+  touch ctestS.done
 else
```
Comment (lrknox) [inline bin/batch/ctest.qsub.in.cmake:18]: There are changes in this PR unrelated to spelling check and spelling corrections.  
No. 1 CMake change to add scripts that append test results when uploading to CDash

File: c++/src/H5AbstractDs.cpp [hunk 2]
```diff
@@ -302,7 +302,7 @@ AbstractDs::getStrType() const
 
 //--------------------------------------------------------------------------
 // Function:    AbstractDs::getVarLenType
-///\brief       Returns the floating-point datatype of this abstract dataset,
+///\brief       Returns the variable length datatype of this abstract dataset,
 ///             which can be a dataset or an attribute.
```
Comment (lrknox) [inline c++/src/H5AbstractDs.cpp:306]: No. 2 Correction to doxygen comment

File: config/cmake/HDFFortranCompilerFlags.cmake [hunk 3]
```diff
@@ -62,7 +62,7 @@ if (NOT MSVC AND NOT MINGW)
   # General flags
   if (CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
     ADD_H5_FLAGS (HDF5_CMAKE_Fortran_FLAGS "${HDF5_SOURCE_DIR}/config/intel-warnings/ifort-general")
-    list (APPEND HDF5_CMAKE_Fortran_FLAGS "-stand f03" "-free")
+    list (APPEND HDF5_CMAKE_Fortran_FLAGS "-stand:f03" "-free")
   elseif (CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
```
Comment (lrknox) [inline config/cmake/HDFFortranCompilerFlags.cmake:66]: No. 3 Add  missing ':' in fortran flag "-stand f03"

File: config/toolchain/aarch64.cmake [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#set (CMAKE_RC_COMPILER ${TOOLCHAIN_PREFIX}-windres)
+set (CMAKE_Fortran_COMPILER ${TOOLCHAIN_PREFIX}-gfortran)
+set (CMAKE_FIND_ROOT_PATH /usr/${TOOLCHAIN_PREFIX})
+set (CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set (CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set (CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set (CMAKE_CROSSCOMPILING_EMULATOR qemu-aarch64)
+
+include_directories(/usr/${TOOLCHAIN_PREFIX}/include)
+
```
Comment (lrknox) [inline config/toolchain/aarch64.cmake:17]: No. 4 Add new file config/toolchain/aarch64.cmake
No. 5 toolchain lowercase names and updates

File: fortran/src/H5f90global.F90 [hunk 5]
```diff
@@ -18,7 +18,7 @@
 !  then the PARAMETER of that corresponding array in Fortran must also be changed.
 !
 ! USES
-!  H5FORTRAN_TYPES 	 - This module is generated at run time. See
+!  H5FORTRAN_TYPES - H5FORTRAN_TYPES is generated at run time by H5match_types.c
 !
```
Comment (lrknox) [inline fortran/src/H5f90global.F90:22]: No. 9 corrected comment in fortran/src/H5f90global.F90

File: src/H5Tpublic.h [hunk 6]
```diff
@@ -1081,7 +1081,7 @@ H5_DLLVAR hid_t H5T_NATIVE_UINT_FAST64_g;
  *          When creating a fixed-length string datatype, \p size will
  *          be the length of the string in bytes. The length of the
  *          string in characters will depend on i the encoding used; see
- *          H5Pset_char_encoding().
+ *          #H5Pset_char_encoding.
```
Comment (lrknox) [inline src/H5Tpublic.h:1084]: This change not in develop.

File: testpar/t_pflush1.c [hunk 7]
```diff
@@ -215,6 +215,7 @@ main(int argc, char *argv[])
     HDfflush(stderr);
     HDprintf("*** ERROR ***\n");
     HDprintf("THERE WAS A REAL ERROR IN t_pflush1.\n");
+    HDfflush(stdout);
```
Comment (lrknox) [inline testpar/t_pflush1.c:218]: No. 10 This call to HDfflush(stdout); is not in the develop branch.   Why is it being added here?

File: config/toolchain/intel_classic.cmake [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+#set(CMAKE_SYSTEM_NAME Linux)
+
+set(CMAKE_COMPILER_VENDOR "intel")
+
+set(CMAKE_C_COMPILER icc)
+set(CMAKE_CXX_COMPILER icpc)
+set(CMAKE_Fortran_COMPILER ifort)
+
+# the following is used if cross-compiling
+set(CMAKE_CROSSCOMPILING_EMULATOR "")
```
Comment (lrknox) [inline config/toolchain/intel_classic.cmake:11]: No. 6 Add new file config/toolchain/intel_classic.cmake.
No. 7 updated copyright header in examples/h5_debug_trace.c
No. 8 updated copyright headers in fortran/examples files.

File: config/toolchain/aarch64.cmake (hunk 4 above)
Comment (byrnHDF) [reply config/toolchain/aarch64.cmake:17]: If the outstanding 1.10 PRs were merged - these wouldn't be a problem

File: testpar/t_pflush1.c (hunk 7 above)
Comment (byrnHDF) [reply testpar/t_pflush1.c:218]: I didn't compare to dev, I think 1.12.

File: bin/batch/ctest.qsub.in.cmake (hunk 1 above)
Comment (byrnHDF) [reply bin/batch/ctest.qsub.in.cmake:18]: These HCP changes should have been in a long time ago, If a PR could be made for these then these files would disappear.

File: c++/src/H5AbstractDs.cpp (hunk 2 above)
Comment (byrnHDF) [reply c++/src/H5AbstractDs.cpp:306]: This is an obvious spelling error! var len does start with a "f".

File: config/cmake/HDFFortranCompilerFlags.cmake (hunk 3 above)
Comment (byrnHDF) [reply config/cmake/HDFFortranCompilerFlags.cmake:66]: Again this is a change that should have been in a PR merge. It is necessary for compiling with Intel.

File: config/toolchain/aarch64.cmake (hunk 4 above)
Comment (byrnHDF) [reply config/toolchain/aarch64.cmake:17]: CI needs to work properly and this is an error change.

File: config/toolchain/intel_classic.cmake (hunk 8 above)
Comment (byrnHDF) [reply config/toolchain/intel_classic.cmake:11]: copyright headers were wrong!

File: fortran/src/H5f90global.F90 (hunk 5 above)
Comment (byrnHDF) [reply fortran/src/H5f90global.F90:22]: Sort of a spelling error.

**Review by lrknox — APPROVED** [review]

**Pull request merged**